CAP/gap/UniversalObjects.gi

```diff
@@ -421,6 +438,20 @@ InstallMethod( MorphismBetweenDirectSums,
 end );
 
 ##
+InstallMethod( MorphismBetweenDirectSums,
+               [ IsCapCategoryObject, IsList, IsCapCategoryObject ],
+               
+  function( S, morphism_matrix, T )
+    
+    if IsZero( S ) or IsZero( T ) then
```

Comment:
Depending on the category this test might be way too expensive
(e.g. in the case of sheaves on toric varieties).
Doesn't it suffice to check if the given list is either empty
or a list containing only empty lists?